js-OOP/spreadAndRest.js

```diff
@@ -0,0 +1,152 @@
+//task 1
+const firstArray = [1, 2, 3];
+const secondArray = [4, 5, 6];
+
+const thirdArray = [...firstArray, ...secondArray];
+console.log(thirdArray);
+
+//task 2
+const multiply = (miltiplier, ...arguments) => arguments.map(argument => argument * miltiplier)
+console.log(multiply(4, 9, 83, 8));
+
+//task 3
+class ProductClass {
+    constructor(name, price, quantity) {
+        this.name = name;
+        this.price = price;
+        this.quantity = quantity;
+    }
+
+    getTotalPrice() {
+        return this.price * this.quantity;
+    }
+
+    purchase(quantityToBuy) {
+        if (quantityToBuy <= this.quantity) {
+            this.quantity -= quantityToBuy;
+            console.log(`The purchase of ${quantityToBuy} units of ${this.name} has been completed`);
+        } else {
+            console.log(`There is not enough product ${this.name} in stock to purchase.`);
+        }
+    }
+}
+
+const tShirt = new ProductClass('T-Shirt', 20, 100);
+const jeans = new ProductClass('Jeans', 30, 50);
+const shoes = new ProductClass('Shoes', 50, 30);
+
+console.log(`Total price of all T-Shirts: $${tShirt.getTotalPrice()}`);
+console.log(`Total price of all jeans: $${jeans.getTotalPrice()}`);
+console.log(`Total price of all shoes: $${shoes.getTotalPrice()}`);
+
+tShirt.purchase(10);
+jeans.purchase(60);
+shoes.purchase(5);
+
+console.log(`Remaining t-shirts in stock: ${tShirt.quantity}`);
+console.log(`Remaining jeans in stock: ${jeans.quantity}`);
+console.log(`Remaining shoes in stock: ${shoes.quantity}`);
+
+//task 4
+class CarClass {
+    constructor(make, model, year) {
+        this.make = make;
+        this.model = model;
+        this.year = year;
+        this.speed = 0;
+    }
+
+    getCarInfo() {
+        console.log(`Make: ${this.make}, Model: ${this.model}, Year: ${this.year}`);
+    }
+
+    accelerate(speed) {
+        this.speed += speed;
+    }
+
+    brake(speed) {
+        this.speed -= speed;
```

Comment:
```suggestion
        this.#speed -= speed;
```

to show this result you can wrap your calculation by console.log(this.#speed -= speed)